feat(tui2): add multi-click transcript selection

joshka-oai opened the pull request:
Support multi-click transcript selection using transcript/viewport coordinates
(wrapped visual line index + content column), not terminal buffer positions.

Gestures:
- double click: select word-ish token under cursor
- triple click: select entire wrapped line
- quad click: select paragraph (contiguous non-empty wrapped lines)
- quint+ click: select the entire history cell (all wrapped lines belonging to a
  single `HistoryCell`, including blank lines inside the cell)

Selection expansion rebuilds the wrapped transcript view from
`HistoryCell::display_lines(width)` so boundaries match on-screen wrapping during
scroll/resize/streaming reflow. Click grouping is resilient to minor drag jitter
(some terminals emit tiny Drag events during clicks) and becomes more tolerant as
the sequence progresses so quad/quint clicks are practical.

Tests cover expansion (word/line/paragraph/cell), sequence resets (timing, motion,
line changes, real drags), drag jitter, and behavior on spacer lines between
history cells (paragraph/cell selection prefers the cell above).

joshka-oai committed:
feat(tui2): add multi-click transcript selection

Support double/triple/quad click selection (word/line/paragraph) using
transcript/viewport coordinates rather than terminal buffer positions.

Multi-click expansion rebuilds the wrapped transcript view from
HistoryCell::display_lines(width) so boundaries match on-screen wrapping
during scroll/resize/streaming reflow. Drag selection resets the click
tracker to avoid accidental multi-click accumulation.

Tests cover click expansion, resets (time, movement, line change, drag),
and paragraph detection across spacer lines between history cells.

joshka-oai committed:
feat(tui2): improve transcript copy fidelity

Improve clipboard output for transcript selections, even when content is wrapped.

- Preserve meaningful indentation for code blocks.
- Treat soft-wrapped prose as a single logical line via wrap joiners.
- Emit Markdown source markers when copying (inline backticks, code fences).
- Extract selection/copy reconstruction into `transcript_copy`.
- Thread joiner metadata through history rendering and wrapping helpers.
- Update docs and add focused test coverage.

joshka-oai committed:
fix(tui2): improve multi-click selection reliability

Multi-click selection was flaky because we reset the click tracker on any mouse
drag event. Some terminals emit Drag events for very small cursor motion while
the button is held down (trackpad jitter), which breaks double/quad click
sequences.

Treat small drags as jitter: only reset the click tracker once the cursor moves
to another wrapped line or far enough horizontally from the anchor. Also loosen
click grouping as the sequence progresses and slightly increase the inter-click
timeout so quad-clicks are practical.

Add tests covering drag jitter and relaxed quad-click grouping.

joshka-oai committed:
feat(tui2): select whole cell on quint-click

Add a 5+ click gesture to select the entire history cell in the transcript.

Implementation notes:
- Selection is computed in transcript/viewport coordinates (wrapped visual line
  indices + content columns), not terminal buffer coordinates.
- We rebuild the wrapped transcript view and carry forward a mapping from each
  wrapped line to its originating HistoryCell index.
- Quint-click expands to the contiguous wrapped-line range for that cell; if the
  click lands on a spacer line between cells, it selects the cell above (falling
  back to the next cell below).

Tests cover selecting the full cell (including blank lines inside the cell) and
quint-click behavior on spacer lines.

bolinfest commented on codex-rs/tui2/src/history_cell.rs:
```diff
@@ -58,6 +56,47 @@ use std::time::Instant;
 use tracing::error;
 use unicode_width::UnicodeWidthStr;
 
+#[derive(Debug, Clone)]
```
I feel like docstring usually appears before `derive`?

joshka-oai committed:
test(tui2): narrow disallowed_methods suppression

The SGR writer tests intentionally construct truecolor (RGB) and indexed colors
so the emitted escape sequences are deterministic.

Replace the blanket allow on the whole test with statement-level
`#[expect(clippy::disallowed_methods)]` on the specific constructors
(`Color::Rgb`/`Color::Indexed`), with inline rationale.
